tests/test_self_update.py

```
@@ -359,3 +401,446 @@ def test_state_roundtrip(tmp_path, monkeypatch):
+    assert calls == {"prepare": 0, "legacy": 1}
+
+
+def test_server_activates_before_engine_imports():
+    # Phase A must run before server.py's `from src.engine...` imports, which
+    # eagerly load the vector stores into memory at module scope. Guard the
+    # ordering invariant statically (numpy-free) so a future reorder is caught.
+    # Match real code lines, not comments/docstrings that mention the strings.
+    server_py = Path(__file__).resolve().parents[1] / "src" / "server.py"
+    lines = server_py.read_text().splitlines()
+    act_line = next(
+        (i for i, l in enumerate(lines)
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)